components/api_request_helper/api_request_helper.cc

```diff
@@ -317,18 +332,40 @@ void APIRequestHelper::URLLoaderHandler::send_sse_data_for_testing(
     std::string_view string_piece,
     bool is_sse,
     DataReceivedCallback callback) {
-  is_sse_ = true;
+  is_sse_ = is_sse;
   data_received_callback_ = std::move(callback);
   OnDataReceived(string_piece, base::BindOnce([]() {}));
 }
 
-data_decoder::DataDecoder*
-APIRequestHelper::URLLoaderHandler::GetDataDecoder() {
+void APIRequestHelper::URLLoaderHandler::ParseJsonImpl(
+    std::string json,
+    data_decoder::DataDecoder::ValueParseCallback callback) {
```

Comment:
what happens if the caller exits before the decode task has completed, is the callback gonna be called anyway?
Is it guaranteed to exists here?

Reply:
the callback will still be called. It's the default behavior. And the caller responsibility to use the correct callback (usually binding with WeakPtr should be used).

BTW, nothing is changed in the public interface, `ParseJsonImpl` is a private thing.